server/routes/settings/index.ts

```diff
@@ -225,6 +225,50 @@ settingsRoutes.post('/plex/sync', (req, res) => {
   return res.status(200).json(plexFullScanner.status());
 });
 
+settingsRoutes.get(
+  '/plex/users',
+  isAuthenticated(Permission.MANAGE_USERS),
+  async (req, res) => {
+    const userRepository = getRepository(User);
+    const qb = userRepository.createQueryBuilder('user');
+
+    const admin = await userRepository.findOneOrFail({
+      select: ['id', 'plexToken'],
+      order: { id: 'ASC' },
+    });
+    const plexApi = new PlexTvAPI(admin.plexToken ?? '');
+    const plexUsers = (await plexApi.getUsers()).MediaContainer.User.map(
+      (user) => user.$
+    );
+
+    const unimportedUsers: {
+      id: string;
+      title: string;
+      username: string;
+      email: string;
+      thumb: string;
+    }[] = [];
+
+    await Promise.all(
+      plexUsers.map(async (user) => {
+        if (
+          !(await qb
+            .where('user.plexId = :id', { id: user.id })
+            .orWhere('user.email = :email', {
+              email: user.email.toLowerCase(),
+            })
+            .getOne()) &&
```

Comment:
I wonder if we can check all the users at once with an `IN` instead of running a ton of queries like this?